### PR TITLE
Add OnnxToTorch support for Compress op

### PR DESCRIPTION
**Pull request by jinchen62**

### CODE REVIEW

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp [hunk 1]
```diff
@@ -704,6 +704,70 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
...
+            numElements *= i;
+          }
+          SmallVector<int64_t> flattenShape = {numElements};
+          auto flattenType = Torch::ValueTensorType::get(
+              binder.op->getContext(), flattenShape, resultType.getDtype());
+          Value flattenTensor = rewriter.create<Torch::AtenFlattenUsingIntsOp>(
+              binder.getLoc(), flattenType, operand, cstZero, cstNegOne);
+          rewriter.replaceOpWithNewOp<Torch::AtenIndexSelectOp>(
+              binder.op, resultType, flattenTensor, cstZero, indexVal);
+        } else {
```
Comment (ScottTodd) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp:761]: Could return early here to reduce nesting.

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp [hunk 2]
```diff
@@ -704,6 +704,70 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
...
+          if (axis < 0)
+            axis += shapeSizes.size();
```
Comment (ScottTodd) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp:763]: Can you link to a spec and/or add a comment here to explain what this is doing? It isn't clear to me what a non-zero axis should mean (maybe this wraps around like a python list index?).

Comment (jinchen62) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp]: Added a comment in code. Negative axis value means counting dimensions from the back.

**Review by ScottTodd — APPROVED** [review]
Seems reasonable to me.

**Comment by vivekkhandelwal1** [comment]
@jinchen62 Did you verify if it's lowering to Linalg or not?

**Pull request merged**